ai/core/src/unitycatalog/ai/core/utils/callable_utils.py

```diff
@@ -769,3 +772,266 @@ def check_docstring_signature_consistency(
             UserWarning,
             stacklevel=2,
         )
+
+
+def _recursively_parse_type(type_info: Any) -> str:
+    """
+    Recursively parse a SQL type definition (provided either as a dict or a string)
+    into a Python type annotation string.
+
+    Args:
+        type_info: A dict containing the type information, or a simple string.
+
+    Returns:
+        A string representing the Python type.
+    """
+    if not isinstance(type_info, dict):
```

Comment:
type_info could be `array<int>`?

Reply:
Yeah on further examination, I think I'm just going to use the SQL full type string directly and remove all of this json configuration parsing. It's redundant, overly complex, and doesn't work for OSS. 
I've deleted the 3 functions involved in the JSON type declaration and simplified all of the approaches to use the full sql type string, recursing over collection types. 
Updated the OSS client tests to validate functionality with proper header signature type extraction as well. I think we're in a better spot now :D